proto/public/api.proto

```diff
@@ -124,8 +124,68 @@ message ApiError {
   string error = 1;
 }
 
+message ResourceSpec {
+  string namespace = 1;
+  string resource = 2; // k8s resource e.g. deploy/foo, or 'all'
+}
+
+message ResourceSelection {
+  ResourceSpec spec = 1;
+  string label_selector = 2;
+}
+
+message ResourceError {
+  ResourceSpec spec = 1;
+  string error = 2;
+}
+
+message StatSummaryRequest {
+  ResourceSelection resource = 1;
+  TimeWindow time_window = 2;
+
+  oneof outbound {
+    Empty none = 3;
+    ResourceSpec out_to_resource   = 4;
+    ResourceSpec out_from_resource = 5;
+  }
+}
+
+message StatSummaryResponse {
+  oneof response {
+    Ok ok = 1;
+    ResourceError error = 2;
+  }
+
+  message Ok {
+    repeated StatTable stat_tables = 1;
+  }
+}
+
+message StatTable {
+  oneof table {
+    PodGroup pod_group = 1;
+  }
+
+  message PodGroup {
+    repeated Row rows = 1;
+
+    message Row {
+      ResourceSpec spec = 1;
+
+      uint64 meshed_pod_count = 2;
+      uint64 total_pod_count = 3;
+
+      uint64 success_count = 4;
+      uint64 failure_count = 5;
+      uint64 latency_ms_p50 = 6;
+      uint64 latency_ms_p90 = 7;
```

Comment:
Related to @briansmith's comment, these 4 stats feel like they'll be applicable to all (or almost all) resources. Maybe we could move them to a separate message that can be reused. Something like:

```
message BasicStats {
  uint64 success_count = 1;
  uint64 failure_count = 2;
  uint64 latency_ms_p50 = 3;
  uint64 latency_ms_p90 = 4;
}
```

And then here:

```
    message Row {
      Resource resource = 1;

      uint64 meshed_pod_count = 2;
      uint64 total_pod_count = 3;

      BasicStats stats = 4;
    }
```